src/int/div.rs

```diff
@@ -0,0 +1,314 @@
+//! [`Int`] division operations.
+
+use core::ops::Div;
+
+use subtle::{Choice, CtOption};
+
+use crate::{CheckedDiv, ConstChoice, Int, NonZero, Uint};
+
+impl<const LIMBS: usize> Int<LIMBS> {
+    #[inline]
+    /// Base checked_div_mod operation.
+    /// Given `(a, b)` computes the quotient and remainder of the absolute
+    /// Note: this operation rounds towards zero, truncating any fractional part of the exact result.
+    fn checked_div_mod(
```

Comment:
can we instead have a method that is consistent with primitive div/rem i.e. return quotient and remainder with quotient round towards and both being signed integer as they both can be negative e.g. -5 / 2 = (-2, -1).
Also to be consistent with Uint can we rename it to div_rem instead of div_mod?

Reply:
Sorry, I was a little chaotic,
What I'm trying to say is that returning two Uints and one bit is not enough to cover all the cases:
```
-5/-2 = (2, -1)
-5/2 = (-2, -1)
5/-2 = (-2, 1)
5/2 = (2, 1)
```

Reply:
Ahh, didn't think of that. Let me fix this later today!

Reply:
I updated the code. Let me know what you think!

Reply:
LGTM